File: dawn/src/main/java/com/lzq/dawn/util/thread/SimpleTask.java
package com.lzq.dawn.util.thread;

import android.util.Log;

/**
 * @Name :SimpleTask
 * @Time :2022/9/2 14:46
 * @Author :  Lzq
 * @Desc :
 */
public abstract class SimpleTask<T> extends Task<T> {

    @Override
    public void onCancel() {
        Log.e("ThreadUtils", "onCancel: " + Thread.currentThread());
    }

    @Override
    public void onFail(Throwable t) {
        Log.e("ThreadUtils", "onFail: ", t);
    }

}

File: dawn/src/main/java/com/lzq/dawn/util/thread/SimpleTask.kt
package com.lzq.dawn.util.thread

import android.util.Log

/**
 * @Name :SimpleTask
 * @Time :2022/9/2 14:46
 * @Author :  Lzq
 * @Desc :
 */
abstract class SimpleTask<T> : Task<T>() {
    override fun onCancel() {
        Log.e("ThreadUtils", "onCancel: " + Thread.currentThread())
    }

    override fun onFail(t: Throwable) {
        Log.e("ThreadUtils", "onFail: ", t)
    }
}